proposals/0000-hover-style-ipad-api.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Hover style API for iPad / Apple Vision Pro
+author:
+  - Oskar Kwaśniewski
+  - Saad Najmi
+date: 08.01.2023
+---
+
+# RFC0000: Hover style API for iPad / Apple Vision Pro
+
+## Summary
+
+iPad supports hover style effects when using an external mouse/trackpad. This effect illuminates the UIView in two configurable ways: lift or highlight. This proposal intends to add support for this API in React Native. This API is also used by visionOS to display the hover effect when a user looks at the view.
+
+The native API this feature is basing on: https://developer.apple.com/documentation/uikit/uihoverstyle (iOS 17+). There is an older version of this API supporting iOS 13.4+ https://developer.apple.com/documentation/uikit/uipointerstyle
+
+## Basic example
+
+This API would introduce an additional prop to UIView (which is used to render touchable/pressable elements).
+
+```jsx
+<Pressable hoverEffect=”lift” onPress={onPressFunction}>
+  <Text>I'm pressable!</Text>
+</Pressable>
```

Comment:
Having unprefixed, single platform props for shared components like `Pressable` is a little bit smelly.

There are various schemes for prefixing and suffixing platform specific props in RN, and also ones we can do that we haven't tried yet (`<Pressable apple:hoverEffect="lift">`)

This might actually belong on the Platform's View (or `VisionOSView` instead), if we would ever want to support hover style outside Pressables.

Reply:
In my head, something like `hoverEffectIOS` o hoverEffectApple` is fine, though I feel it has been somewhat inconsistent in the past ( `elevation` isn't `elevationAndroid`, etc..). As implemented in the visionOS fork right now, it is a prop on the base `<View>`, and `<Pressable>` just sets a default that isn't `none`.

Reply:
I’m curious, what do you call these events for Windows Ink (Surface or Asus Pen)? Or for hand selection tracking in MRTK/HoloLens? I think there’s both prior art and probably a way to (finally) pull some of these offshoots forward in a unified way.

Reply:
I have no idea what windows would call it 😅

Reply:
RN Windows solution for this is pretty crummy.

The event shape for custom `onMouseEnter`, `onMouseLeave` is effectively exact copy of [`PointerPoint`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.input.pointerpoint?view=windows-app-sdk-1.4)

So, API is not coherent with other platforms. IIRC it may not even be typed.

Reply:
Pointer events unifies all of that, for I guess everything but Vision OS requiring declarative handling.